### DISCUSSION TITLE
How do I customise the BibTeX output format?

### DISCUSSION THREAD

**Discussion by chrisvoncsefalvay** [quarto-dev/quarto-cli#7234] [Q&A] [answered]
### Description

I love Quarto, but the BibTeX output format and the citation reference pose two problems with presenting my name. My last name has a particle (`von`) and a last name. This creates two issues:

1. BibTeX cite key generation: the autogenerated cite key has a space in it, between the particle and the last name. This is of course going to break the cite key usage.
2. Referencable posts: here, the particle is treated as a first name rather than part of the last name.

How do I configure these citation renderers?

**Comment by mcanouil** [comment]
See <https://quarto.org/docs/authoring/footnotes-and-citations.html> in particular the citation style section.

If you actually talking about the citation entry added via `citation: true`, see <https://quarto.org/docs/reference/metadata/citation.html>.

For more help, please clarify and possibly provide an (reproducible) example.

**Comment by chrisvoncsefalvay** [comment]
Reproducible example:

Input: https://github.com/chrisvoncsefalvay/chrisvoncsefalvay.github.io/blob/main/posts/llms-language/index.qmd

Output: https://chrisvoncsefalvay.com/posts/llms-language/

The question here, really, is whether there's any alternative to change how the boxes at the bottom for the `citation: true` entries are rendered by default, i.e. without having to manually specify everything in every post. This is the current display for the BibTeX citation:

```
@online{von csefalvay2023,
  author = {von Csefalvay, Chris},
  title = {Beyond {Broca}},
  date = {2023-10-15},
  url = {https://chrisvoncsefalvay.com/posts/llms-language},
  langid = {en}
}
```

I'd like the cite key to not have a space in it, obviously.

This is the reference:

```
Csefalvay, Chris von. 2023. “Beyond Broca.” October 15, 2023. https://chrisvoncsefalvay.com/posts/llms-language.
```

This should read

```
von Csefalvay, Chris. 2023....
```

**Reply by mcanouil** [reply, marked as answer]
If the bibtex entry is as you wish, which seems to be the case, then how the citation is displayed depends on the CSL style, see my previous answer.

Also, you should give Quarto the appropriate details about your name according to the authors schema, see <https://quarto.org/docs/journals/authors.html#author-schema>.

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/99a59f9a-ed6c-4933-8de1-fb11768268bf">

**Comment by chrisvoncsefalvay** [comment]
OK, so the answer ended up being using a `_metadata.yml` file for the whole `/posts` directory. Thanks!

**Reply by mcanouil** [reply]
I am not sure to follow, but indeed if you are the only author of your website, it is best to define author and common metadata in that file instead of in every posts.

No matter where you defined your name, you still need to define it appropriately and to use the style that suits you.
